lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/IntegrationExpansion.cxx

```diff
@@ -0,0 +1,195 @@
+    const Distribution & distribution)
+  : IntegrationExpansion(inputSample, Point(inputSample.getSize(), 1.0), outputSample, distribution)
+{
+  // Nothing to do  
+}
+
+/* Constructor */
+IntegrationExpansion::IntegrationExpansion(const Sample & inputSample,
+    const Point & weights,
+    const Sample & outputSample,
+    const Distribution & distribution)
+  : FunctionalChaosAlgorithm(inputSample, weights, outputSample, distribution)
+{
+  // Extract the basis from the adaptive strategy attribute of the base class
+  basis_ = getAdaptiveStrategy().getBasis();
+  // The basis size is the maximum dimension of the adaptive strategy
+  basisSize_ = getAdaptiveStrategy().getMaximumDimension();
+}
+
+/* Constructor */
+IntegrationExpansion::IntegrationExpansion(const Sample & inputSample,
+    const Sample & outputSample,
+    const Distribution & distribution,
+    const OrthogonalBasis & basis,
+    const UnsignedInteger basisSize)
+  : IntegrationExpansion(inputSample, Point(inputSample.getSize(), 1.0), outputSample, distribution, basis, basisSize)
+{
+  // Nothing to do  
+}
+
+/* Constructor */
+IntegrationExpansion::IntegrationExpansion(const Sample & inputSample,
+    const Point & weights,
+    const Sample & outputSample,
+    const Distribution & distribution,
+    const OrthogonalBasis & basis,
+    const UnsignedInteger basisSize)
+  : FunctionalChaosAlgorithm(inputSample, weights, outputSample, distribution)
+  , basis_(basis)
+  , basisSize_(basisSize)
+{
+  // The arguments are checked in the base class, excepted the basis and its size
+  if (basis.getMeasure().getDimension() != distribution.getDimension()) throw InvalidArgumentException(HERE) << "Error: the basis must have a measure with the same dimension as the input distribution, heare measure dimension=" << basis.getMeasure().getDimension() << " and distribution dimension=" << distribution.getDimension();
+  if (basisSize == 0) throw InvalidArgumentException(HERE) << "Error: cannot project on a basis of size zero";
+}
+
+
+/* Virtual constructor */
+IntegrationExpansion * IntegrationExpansion::clone() const
+{
+  return new IntegrationExpansion(*this);
+}
+
+
+/* String converter */
+String IntegrationExpansion::__repr__() const
+{
+  return OSS() << "class=" << getClassName();
+}
+
+
+/* Computes the functional chaos */
+void IntegrationExpansion::run()
+{
+  // Check if we have to transform the input sample in order to adapt it
+  // to the basis
+  const Distribution measure(basis_.getMeasure());
+  const Bool noTransformation = (distribution_ == measure);
+  Sample transformedInputSample;
+  DistributionTransformation transformation;
+  if (noTransformation) transformedInputSample = inputSample_;
+  else
+    {
+      transformation = DistributionTransformation(distribution_, basis_.getMeasure());
+      transformedInputSample = transformation(inputSample_);
+    }
+  // Build the relevant functions from the basis
+  Indices indices(basisSize_);
+  indices.fill();
```

Comment:
I suggest to provide a way to directly set the indices when they are known. Please see my comment on `LeastSquaresExpansion`.

Reply:
It will be done through the addition of a `setActiveFunctions()` method.